core/src/main/java/com/microsoft/applicationinsights/internal/agent/CoreAgentNotificationsHandler.java

```diff
@@ -1,514 +1,520 @@
-                t.printStackTrace();
-            }
-        }
-    }
-
-    private static long nanoToMilliseconds(long nanoSeconds) {
-        return nanoSeconds / 1000000;
-    }
-
-    private StringBuilder formatAdditionalSqlArguments(MethodData methodData) {
-        try {
-            StringBuilder sb = new StringBuilder();
-            sb.append(" [");
-            Object[] args = (Object[])methodData.arguments[3];
-            if (args != null && args.length > 0) {
-                for (Object arg : args) {
-                    if (arg == null) {
-                        sb.append("null,");
-                    } else {
-                        sb.append(arg.toString());
-                        sb.append(',');
-                    }
-                }
-                sb.deleteCharAt(sb.length() - 1);
-            }
-            sb.append(']');
-            return sb;
-        } catch (Throwable t) {
-            return null;
-        }
-    }
-
-    private StringBuilder fetchExplainQuery(String commandName, Object object) {
-        StringBuilder explainSB = null;
-        Statement explain = null;
-        ResultSet rs = null;
-        try {
-            if (commandName.startsWith("SELECT ")) {
-                Connection connection = (Connection)object;
-                if (connection == null) {
-                    return explainSB;
-                }
-                explain = connection.createStatement();
-                rs = explain.executeQuery("EXPLAIN " + commandName);
-                explainSB = new StringBuilder();
-                while (rs.next()) {
-                    explainSB.append('[');
-                    int columns = rs.getMetaData().getColumnCount();
-                    if (columns == 1) {
-                        explainSB.append(rs.getString(1));
-                    } else {
-                        for (int i1 = 1; i1 < rs.getMetaData().getColumnCount(); ++i1) {
-                            explainSB.append(rs.getMetaData().getColumnName(i1));
-                            explainSB.append(':');
-                            Object obj = rs.getObject(i1);
-                            explainSB.append(obj == null ? "" : obj.toString());
-                            explainSB.append(',');
-                        }
-                        explainSB.deleteCharAt(explainSB.length() - 1);
-                    }
-                    explainSB.append("],");
-                }
-                explainSB.deleteCharAt(explainSB.length() - 1);
-            }
-        } catch (Throwable t) {
-        } finally {
-            if (rs != null) {
-                try {
-                    rs.close();
-                } catch (SQLException e) {
-                }
-            }
-            if (explain != null) {
-                try {
-                    explain.close();
-                } catch (SQLException e) {
-                }
-            }
-        }
-
-        return explainSB;
-    }
-}
+/*

```

Comment:
what changed in this file?

Reply:
@grlima to comment more on this. At-least my intuition here was that while sending telemetry for SqlServer there are only two arguments that come in the methoddata object whereas previously there was a strict condition which checked arguments should be more than or equal to 3

Reply:
@littleaj this file was edited by @grlima when prepared statements for sqlserver where not working (i.e not capturing the query data) we can no revert this changes back because we no longer need to loose if condition in the sendSQLTelemetry() method.

Reply:
you can see the difference here: `git diff master -- CoreAgentNotificationsHandler.java > CoreAgentNotificationsHandler.java.diff` (assuming PWD is the file's location). It's the line endings. Your IDE should be set to preserve the existing line ending characters. There may be a way to fix this in IntelliJ, but you may have to load it in Notepad++ an do a find/replace.

If you want to just checkout a clean copy and make the changes again, add use this command to see the "actual" changes: `git diff master -w -- CoreAgentNotificationsHandler.java > CoreAgentNotificationsHandler.java.diff` to ignore all whitespace changes